### PR TITLE
added sentry error logging to api

### PR DESCRIPTION
**Pull request by MwayaB**
**Description**

Added sentry error logging to the backend.

Jira link:

https://lite-farm.atlassian.net/browse/LF-3359

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: '3.7'
+version: "3.7"
```
Comment (iperdomo) [inline docker-compose.yml:1]: This are unrelated changes. Can we revert them?

File: packages/api/src/server.js [hunk 2]
```diff
@@ -23,10 +23,35 @@ import './dotenvConfig.js';
...
+
+    // Set tracesSampleRate to 1.0 to capture 100%
+    // of transactions for performance monitoring.
+    // We recommend adjusting this value in production
+    tracesSampleRate: 1.0,
+  });
+
+  // RequestHandler creates a separate execution context, so that all
+  // transactions/spans/breadcrumbs are isolated across requests
+  app.use(Sentry.Handlers.requestHandler());
```
Comment (iperdomo) [inline packages/api/src/server.js:51]: Why do we want `requestHandler` and `tracingHandler`? and not only `errorHandler`?

**Review by iperdomo — CHANGES_REQUESTED** [review]
* Let's remove the unintended changes in `docker-compose.yml`
* Let's add the needed missing `SENTRY_DSN` env variable to the `.env.default` under `packages/api`
* Document why do we want/need tracing and request handlers

File: docker-compose.yml (hunk 1 above)
Comment (MwayaB) [reply docker-compose.yml:1]: @iperdomo I think thats the linter

File: packages/api/src/server.js (hunk 2 above)
Comment (MwayaB) [reply packages/api/src/server.js:51]: @iperdomo to capture detailed performance data, it was suggested in the sentry project creation wizard, i saw no reason to not include it.

File: docker-compose.yml (hunk 1 above)
Comment (kathyavini) [reply docker-compose.yml:1]: Sorry!

**Review by kathyavini — APPROVED** [review]
I am approving the cleaned up diff but I haven't tested the Sentry functionality 🙂 

(Good rebase!)
